crates/chia-sdk-driver/src/primitives/cat.rs

```diff
@@ -414,7 +443,7 @@ impl Cat {
 
         // If the hint wrapped in the revocation layer of the parent matches the create coin's puzzle hash,
         // then we know that the hint is the p2 puzzle hash and the child has the same revocation layer as the parent
-        if hint
+        if create_coin.puzzle_hash
```

Comment:
this seems like a pretty stark change. was this a bug?

Reply:
Yeah, the hint is always the p2 puzzle hash, whereas the revocation layer is the inner puzzle